src/al.jl

```diff
@@ -30,6 +29,15 @@ function al(nlp :: AbstractNLPModel; max_iter :: Int = 1000, max_time :: Real =
                                  iter = iter, elapsed_time = el_time)
   end
 
+  # number of slack variables
+  ns = nlp.meta.ncon - length(nlp.meta.jfix)
+
+  # SlackModel create slack variables if necessary
+  nlp = SlackModel(nlp)
+
+  x = copy(nlp.meta.x0)
+  x = T.(x)
```

Comment:
This can be fixed after https://github.com/JuliaSmoothOptimizers/NLPModels.jl/issues/217, please open an issue so we can remember it.